src/Client/Handlers/IHandler.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace OmniSharp.Extensions.LanguageServerProtocol.Client.Handlers
+{
+    /// <summary>
+    ///     Represents a client-side message handler.
+    /// </summary>
+    public interface IHandler
+    {
+        /// <summary>
+        ///     The name of the method handled by the handler.
+        /// </summary>
+        string Method { get; }
+
+        /// <summary>
+        ///     The kind of handler.
+        /// </summary>
+        HandlerKind Kind { get; }
```

Comment:
`HandlerKind` and the `Kind` property seem unnecessary.  Worst case we can pattern match, but currently they aren't used anywhere but setting them..

Reply:
My original thinking on this was that the handlers themselves would be generic and so it would be hard to pattern-match on them without already knowing the handler payload type (but you're right it's probably redundant now).